services/catarse/engines/catarse_pagarme/app/controllers/catarse_pagarme/pix_controller.rb

```diff
@@ -0,0 +1,59 @@
+require 'weekdays'
+
+module CatarsePagarme
+  class PixController < CatarsePagarme::ApplicationController
+
+    def create
+      transaction = PixTransaction.new(pix_attributes, payment).charge!
+      render json: { pix_qr_code: transaction.pix_qr_code, payment_status: transaction.status, gateway_data: payment.gateway_data }
+    rescue PagarMe::PagarMeError => e
+      raven_capture(e)
+      render json: { pix_qr_code: nil, payment_status: 'failed', message: e.message }
+    end
+
+    def update
+      payment.generating_second_pix = true
+      transaction = PixTransaction.new(pix_attributes, payment).charge!
+      redirect_to main_app.project_contribution_path(contribution.project, contribution)
+    rescue PagarMe::PagarMeError => e
+      raven_capture(e)
+      render json: { pix_qr_code: nil, payment_status: 'failed', message: e.message }
+    end
+
+    def pix_data
+      render json: {pix_expiration_date: payment.pix_expiration_date.to_date}
+    end
+
+    protected
+
+    def pix_attributes
+      {
+        payment_method: 'pix',
+        amount: delegator.value_for_transaction,
+        pix_expiration_date: payment.pix_expiration_date,
+        pix_additional_fields: [
```

Comment:
**Sugestão**
Se não for obrigatório enviar esses campos adicionais, pode remover. E olha se dá pra enviar os campos de metadata que o boleto e o cartão de crédito enviam pelo parâmetro `metadata`. Ficaria assim:
```suggestion
        metadata: metadata_attributes.merge({ second_pix: payment.generating_second_pix.to_s })
```